packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/useHandleUpload.ts

```diff
@@ -80,6 +80,20 @@ export function useHandleUpload({
     }))
   );
 
+  React.useEffect(() => {
+    // update tasks when items changes
+    setTasks(() =>
+      items.map(({ key, data }) => ({
+        cancel: () => setTasks((prev) => removeTask(prev, key)),
+        key,
+        data: data!,
+        status: 'INITIAL',
+        message: undefined,
+        progress: 0,
+      }))
+    );
+  }, [items]);
```

Comment:
Does this run on initial render?

Reply:
Yes I think it does, do we need to stop it from running on the initial render or anything like that?